app/scripts/metamask-controller.js

```diff
@@ -5299,8 +5269,12 @@ export default class MetamaskController extends EventEmitter {
 
     Object.keys(this.connections).forEach((origin) => {
       Object.values(this.connections[origin]).forEach(async (conn) => {
-        if (conn.engine) {
-          conn.engine.emit('notification', await getPayload(origin));
+        try {
+          if (conn.engine) {
+            conn.engine.emit('notification', await getPayload(origin));
+          }
+        } catch (err) {
+          console.error(err);
```

Comment:
why is this try/catch necessary now?

Reply:
in the event that the payload function throws an error

Reply:
oh yeah if the await rejects its unhandled promise rejection